components/rhs_thread/index.js

```diff
@@ -3,21 +3,28 @@
 
 import {connect} from 'react-redux';
 import {bindActionCreators} from 'redux';
-import {getPost, makeGetPostsForThread} from 'mattermost-redux/selectors/entities/posts';
+import {makeGetPostsForThread} from 'mattermost-redux/selectors/entities/posts';
 import {removePost} from 'mattermost-redux/actions/posts';
 
+import {getSelectedPost} from 'selectors/rhs.jsx';
 import RhsThread from './rhs_thread.jsx';
+import {PostTypes} from 'utils/constants.jsx';
 
 function makeMapStateToProps() {
     const getPostsForThread = makeGetPostsForThread();
 
     return function mapStateToProps(state, ownProps) {
-        const selected = getPost(state, state.views.rhs.selectedPostId);
+        const selected = getSelectedPost(state);
+
         let posts = [];
         if (selected) {
             posts = getPostsForThread(state, {rootId: selected.id, channelId: selected.channel_id});
         }
 
+        if (posts.length > 0 && selected.type === PostTypes.FAKE_PARENT_DELETED) {
+            selected.create_at = posts[0].create_at;
```

Comment:
This isn't good since it's mutating the selected post which should be treated as immutable. The best way I can think to work around it would be:
1. When [RhsThread sets the rootPostDay variable in its render method](https://github.com/grundleborg/mattermost-webapp/blob/1d8b4c1ce9a9bb32510316363d9fc90bfb2336e6/components/rhs_thread/rhs_thread.jsx#L354), use posts[0].create_at if selected.create_at isn't set
2. Similarly, add the ability to pass a createAt timestamp into the RhsRootPost component and use the same logic to fall back there

Reply:
Implemented 1. Didn't do 2 as the time tag in the RhsRootPost is not shown for this fake message intentionally, so it doesn't need the create_at value.